### PR TITLE
os/bluestore: pass strict flag to bluestore_blob_use_tracker_t::equal()

### PR DESCRIPTION
**Pull request by xiexingguo**
And if that flag is true, we'll do a strict equivalence comparison instead.

Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by ifed01** [comment]
I'm disagree with the initial assumption that <1> isn't equal to <2>.  In fact <2> doesn't have that actual range information [1..1001] - it contains total referenced bytes counter only. And that amount of referenced bytes equivalent (weakly?) to the one at <1> hence returning true. I.e. equal method returns true if there is a chance that two trackers describe the same used_range rather than strict comparison. Probably it makes sense to change the function name though..

**Comment by xiexingguo** [comment]
> I.e. equal method returns true if there is a chance that two trackers describe the same used_range rather than strict comparison.

I see the only caller is fsck(), which means we use this method to the same blob, so I guess we'd better narrow the constraints down.

**Comment by ifed01** [comment]
I'd prefer to leave current comparison mode as well. May be we can have a boolean flag for equal() call to request strict/weak equivalence comparison.

**Comment by xiexingguo** [comment]
> I'd prefer to leave current comparison mode as well. May be we can have a boolean flag for equal() call to request strict/weak equivalence comparison.

Updated!

File: src/os/bluestore/bluestore_types.cc [hunk 1]
```diff
@@ -488,7 +488,8 @@ void bluestore_blob_use_tracker_t::split(
 }
 
 bool bluestore_blob_use_tracker_t::equal(
-  const bluestore_blob_use_tracker_t& other) const
+  const bluestore_blob_use_tracker_t& other,
```
Comment (ifed01) [inline src/os/bluestore/bluestore_types.cc]: Could you please add some UT for that

Comment (ifed01) [inline src/os/bluestore/bluestore_types.cc]: LGTM.

**Comment by xiexingguo** [comment]
> Could you please add some UT for that

Sure, dded.

**Comment by xiexingguo** [comment]
http://pulpito.ceph.com/xxg-2017-09-01_07:17:41-rados-wip-xxg-testing-2017-09-01-1357-distro-basic-smithi/
http://pulpito.ceph.com/xxg-2017-09-03_12:51:11-rados-wip-xxg-testing-2017-09-01-1357-distro-basic-smithi/

**Comment by xiexingguo** [comment]
retest this please

**Comment by xiexingguo** [comment]
retest this please

**Pull request merged**